library/src/main/java/com/novoda/downloadmanager/lib/DownloadJob.java

```diff
@@ -0,0 +1,45 @@
+package com.novoda.downloadmanager.lib;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+
+import com.evernote.android.job.Job;
+import com.evernote.android.job.JobRequest;
+import com.novoda.downloadmanager.lib.logger.LLog;
+
+import java.util.concurrent.TimeUnit;
+
+class DownloadJob extends Job {
+
+    public static final String TAG = "download_job_tag";
+
+    private static final long BACKOFF_MILLIS = TimeUnit.SECONDS.toMillis(5);
+    private static final long EXECUTION_START_MILLIS = TimeUnit.SECONDS.toMillis(1);
+    private static final long EXECUTION_END_MILLIS = TimeUnit.SECONDS.toMillis(10);
+
+    @NonNull
+    @Override
+    protected Result onRunJob(Params params) {
+        Context context = GlobalState.getContext();
```

Comment:
I have tried to crash it but with no success. What I think that is happening is that the context is held by a static field. Then the jobScheduler is instantiated by the same context so that as long as the job is alive (scheduled) the context will always have a non null reference.